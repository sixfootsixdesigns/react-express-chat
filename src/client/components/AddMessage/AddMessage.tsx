import React from 'react';

interface AddMessageProps {
  addMessage: (message: string) => void;
}

interface AddMessageState {
  message: string;
}

export class AddMessage extends React.Component<
  AddMessageProps,
  AddMessageState
> {
  public state = {
    message: ''
  };

  handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
    this.setState({
      message: event.currentTarget.value
    });
  };

  handleAddMessage = (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    event.preventDefault();
    if (this.state.message) {
      this.props.addMessage(this.state.message);
      this.setState({
        message: ''
      });
    }
  };

  public render() {
    const { message } = this.state;
    return (
      <div className="AddMessageWrapper">
        <div className="AddMessage">
          <textarea
            placeholder="Enter your message"
            onChange={this.handleChange}
            value={message}
          />
          <button onClick={this.handleAddMessage}>Post</button>
        </div>
      </div>
    );
  }
}
