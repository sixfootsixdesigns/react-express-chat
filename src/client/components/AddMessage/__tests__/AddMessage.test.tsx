import * as React from 'react';
import {
  render,
  fireEvent,
  wait,
  getByText,
  getByPlaceholderText,
  cleanup
} from 'react-testing-library';
import { AddMessage } from '../AddMessage';

describe('Add Message Component', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render', () => {
    const addMessageHandler = jest.fn();

    const { container } = render(<AddMessage addMessage={addMessageHandler} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not call addUser if no user is entered', async () => {
    const addMessageHandler = jest.fn();

    const { container } = render(<AddMessage addMessage={addMessageHandler} />);

    const button = getByText(container, 'Post');
    fireEvent.click(button);

    await wait(() => {
      expect(addMessageHandler).not.toHaveBeenCalled();
    });
  });

  it('should call addUser if user is entered', async () => {
    const addMessageHandler = jest.fn();

    const { container } = render(<AddMessage addMessage={addMessageHandler} />);

    const input = getByPlaceholderText(container, 'Enter your message');
    fireEvent.change(input, {
      target: { value: 'hello' }
    });

    const button = getByText(container, 'Post');
    fireEvent.click(button);

    await wait(() => {
      expect(addMessageHandler).toHaveBeenCalledWith('hello');
    });
  });
});
