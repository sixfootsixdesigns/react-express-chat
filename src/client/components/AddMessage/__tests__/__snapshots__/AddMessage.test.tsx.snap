// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Message Component should render 1`] = `
<div
  class="AddMessageWrapper"
>
  <div
    class="AddMessage"
  >
    <textarea
      placeholder="Enter your message"
    />
    <button>
      Post
    </button>
  </div>
</div>
`;
